state/store.go

```diff
@@ -330,7 +341,6 @@ func (store dbStore) batchDelete(start []byte, end []byte, exception []byte) err
 	if err != nil {
 		return fmt.Errorf("iterator error: %w", err)
 	}
-	defer iter.Close()
```

Comment:
Can I remove the defer, or should I still leave it there. I'm just thinking about what happens if we get an error before closing the iterator. Would it still close the iterator?

Reply:
Yeah, we should defer too, otherwise we can end up holding the lock forever if there's an error. This is assuming boltdb or tm-db doesn't mind potentially closing the iterator twice -- if so, we'll have to wrap it in a `sync.Once`, or alternatively restructure the code such that `defer` is always appropriate.